coils_tet_mesh.py

```diff
@@ -0,0 +1,277 @@
+        
+        # Create the mesh
+        cubit.cmd(f'volume {coil_num} scheme tetmesh')
+        cubit.cmd(f'mesh volume {coil_num}')
+
+        # Set log file path and initiate logging
+        log_path = f"{log_prefix}_{coil_num}.{log_ext}"
+        cubit.cmd(f'logging on file "{log_path}"')
+
+        # Run mesh quality metrics for each relevant metric
+        for metric in quality_metric_list:
+            qual_command = f'quality volume {coil_num} {metric} global draw mesh list'
+            mesh_analytics = cubit.cmd(qual_command)
+
+
+        # MESH QUALITY ANALYSIS
+
+        # Read the log file
+        with open(log_path, 'r') as file:
+            text = file.read()
+
+        # Extract the data from the log file using the extract_tabular_data function
+        function_names, averages, std_devs, minima, maxima = extract_tabular_data(text)
+
+        # Separate multi-word function names to be able to match-case with Cubit names
+        function_names = split_function_names(function_names)
+
+        # Compile the extracted tabular data into a Pandas data frame
+        quality_metrics_df = pd.DataFrame({'Function Name': function_names,
+                                           'Average': averages,
+                                           'Standard Deviation': std_devs,
+                                           'Minimum': minima,
+                                           'Maximum': maxima})
+
+        # Remove rows where the standard deviation equals 0
+        # Found this for a few cases and it appeared too implausible to keep
+        quality_metrics_df = quality_metrics_df.loc[quality_metrics_df['Standard Deviation'] != 0]
+
+        # Sort rows by highest average quality metric and select the highest
+        quality_metrics_df = quality_metrics_df.sort_values('Average', ascending=False)
+        best_analytic = quality_metrics_df['Function Name'][0]
+
+        # Override the current mesh analytic with the best quality metric
+        cubit.cmd(f'quality volume {coil_num} {best_analytic} global draw mesh list')
+
+        # Save the best row into the top_results dataframe
+        top_row_df = quality_metrics_df.iloc[0]
+        top_row = list(top_row_df)
+        row_df = pd.DataFrame([top_row], columns=top_results_cols)
+        top_results = pd.concat([top_results, row_df])
+ 
+
+    # MESH EXPORT
+
+    # Export the mesh as an EXODUS file
+    cubit.cmd(f'export mesh "{exo_path}"')
+
+    # Initialize the MOAB core instance
+    mb = core.Core()
+
+    # Load the EXODUS file
+    exodus_set = mb.create_meshset()
+    mb.load_file(exo_path, exodus_set)
+    
+    # Create a new meshset for each coil iteration
+    coil_meshset = mb.create_meshset()
+
+    # Add the current coil's mesh to the meshset
+    mb.add_entities(coil_meshset, [exodus_set])
+
+    # Write the current coil's meshset to an individual .h5m file
+    mb.write_file(h5m_path, [coil_meshset])
```

Comment:
Method to export the mesh - make it optional based on the user input

Reply:
Also update as in the parametric stellarator code

Reply:
Consider a method that can be reused here and in `parametric_stellarator` for mesh export as H5M.